crt_portal/cts_forms/views.py

```diff
@@ -685,12 +681,32 @@ def disposition_view(request):
     disposition_status = request.GET.get('disposition_status', 'past')
     profile_form = get_profile_form(request)
     if disposition_status == 'batches':
+        disposition_batches = ReportDispositionBatch.objects.all()
+        per_page = request.GET.get('per_page', request.COOKIES.get('complaint_view_per_page', 15))
+        page = request.GET.get('page', 1)
+        sort_expr, sorts = other_sort(request.GET.getlist('sort'), 'batch')
+        disposition_batches = disposition_batches.annotate(retention_schedule=Subquery(ReportDisposition.objects.filter(batch=OuterRef("pk")).values_list('schedule', flat=True).distinct()))
+        disposition_batches = disposition_batches.order_by(*sort_expr)
+        paginator = Paginator(disposition_batches, per_page)
+        disposition_batches, page_format = pagination(paginator, page, per_page)
+        sort_state = {}
+        page_args = f'?per_page={per_page}'
+        sort_args, sort_state = get_sort_args(sorts, sort_state)
+        page_args += sort_args
+        filter_args = f'&disposition_status={disposition_status}'
+        page_args += filter_args
+        all_args_encoded = urllib.parse.quote(f'{page_args}&page={page}')
+        data = get_batch_data(disposition_batches)
         final_data = {
             'disposition_status': disposition_status,
+            'page_format': page_format,
+            'page_args': page_args,
+            'per_page': per_page,
+            'sort_state': sort_state,
+            'filter_state': filter_args,
+            'return_url_args': all_args_encoded,
+            'data': data,
```

Comment:
this looks familiar - worth a helper function?

Reply:
I looked into this because I agree this is repetitive but I discovered that there are so many small differences between the instances where this code is used that we wouldn't end up saving that many lines of code ultimately. That said, I think trying to make this functionality more reusable is worth exploring maybe in another PR.